File: app/src/main/java/me/albfan/boogdroid/ui/ServerListFragment.java
/*
 *  BugsControl
 *  Copyright (C) 2014  Jon Ander Peñalba
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package me.albfan.boogdroid.ui;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.ListFragment;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.TextView;

import me.albfan.boogdroid.R;
import me.albfan.boogdroid.general.Server;

public class ServerListFragment extends ListFragment {
    private OnServerSelectedListener listener;
    private ServerTypeAdapter adapter;

    public interface OnServerSelectedListener {
        public void onServerSelected(final int position);
    }

    @Override
    public void onAttach(final Activity activity) {
        super.onAttach(activity);
        listener = (OnServerSelectedListener) activity;
    }

    @Override
    public View onCreateView(final LayoutInflater inflater, final ViewGroup container, final Bundle savedInstanceState) {
        final View view = inflater.inflate(R.layout.product_list_fragment, container, false);

        EditText filterProduct = ((EditText) view.findViewById(R.id.editFilterProduct));
        filterProduct.addTextChangedListener(new TextWatcher() {

            @Override
            public void onTextChanged(CharSequence cs, int arg1, int arg2, int arg3) {
                adapter.getFilter().filter(cs);
            }

            @Override
            public void beforeTextChanged(CharSequence arg0, int arg1, int arg2, int arg3) { }

            @Override
            public void afterTextChanged(Editable arg0) {}
        });

        adapter = new ServerTypeAdapter(getActivity());
        setListAdapter(adapter);
        return view;
    }

    public int getStatusBarHeight(){
        int result = 0;
        int resId = getResources().getIdentifier("status_bar_height", "dimen", "android");
        if(resId > 0){
            result = getResources().getDimensionPixelSize(resId);
        }
        result += ActivityServer.getFakeToolHeight();
        return result;
    }

    @Override
    public void onResume() {
        super.onResume();
        adapter.notifyDataSetChanged();
    }

    @Override
    public void onListItemClick(final ListView l, final View v, final int position, final long id) {
        listener.onServerSelected(position);
    }

    private class ServerTypeAdapter extends ArrayAdapter<Server> {
        public ServerTypeAdapter(final Context context) {
            super(context, R.layout.adapter_server, R.id.name, Server.servers);
        }

        @Override
        public View getView(final int position, View convertView, final ViewGroup parent) {
            if (convertView == null) {
                convertView = getActivity().getLayoutInflater().inflate(R.layout.adapter_server, parent, false);
            }

            final Server s = Server.servers.get(position);

            ((TextView) convertView.findViewById(R.id.name)).setText(s.getName());

            final ImageView iconImage = (ImageView) convertView.findViewById(R.id.icon);
            iconImage.setImageResource(Server.typeIcon.get(Server.typeName.indexOf(s.getType())));

            convertView.findViewById(R.id.delete_button).setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(final View view) {
                    new DialogDeleteServer().setAdapter(adapter).setServerPos(position).show(getFragmentManager(), "DeleteServerDialog");
                }
            });

            return convertView;
        }
    }
}

File: app/src/main/kotlin/ws/lamm/bugdroid/ui/ui/ServerListFragment.kt
package ws.lamm.bugdroid.ui.ui

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.support.v4.app.ListFragment
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.EditText
import android.widget.ImageView
import android.widget.ListView
import android.widget.TextView

import me.johnmh.boogdroid.R
import ws.lamm.bugdroid.general.Server

class ServerListFragment : ListFragment() {
    private var listener: OnServerSelectedListener? = null
    private var adapter: ServerTypeAdapter? = null

    interface OnServerSelectedListener {
        fun onServerSelected(position: Int)
    }

    override fun onAttach(activity: Activity) {
        super.onAttach(activity)
        listener = activity as OnServerSelectedListener?
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.product_list_fragment, container, false)

        val filterProduct = view.findViewById<View>(R.id.editFilterProduct) as EditText
        filterProduct.addTextChangedListener(object : TextWatcher {

            override fun onTextChanged(cs: CharSequence, arg1: Int, arg2: Int, arg3: Int) {
                adapter!!.filter.filter(cs)
            }

            override fun beforeTextChanged(arg0: CharSequence, arg1: Int, arg2: Int, arg3: Int) {}

            override fun afterTextChanged(arg0: Editable) {}
        })

        adapter = activity?.let { ServerTypeAdapter(it) }
        listAdapter = adapter
        return view
    }

    override fun onResume() {
        super.onResume()
        adapter!!.notifyDataSetChanged()
    }

    override fun onListItemClick(l: ListView?, v: View?, position: Int, id: Long) {
        listener!!.onServerSelected(position)
    }

    private inner class ServerTypeAdapter(context: Context) : ArrayAdapter<Server>(context, R.layout.adapter_server, R.id.name, Server.servers) {

        override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
            var convertView = convertView
            if (convertView == null) {
                convertView = activity!!.layoutInflater.inflate(R.layout.adapter_server, parent, false)
            }

            val s = Server.servers[position]

            (convertView!!.findViewById<View>(R.id.name) as TextView).text = s.name

            val iconImage = convertView.findViewById<View>(R.id.icon) as ImageView
            iconImage.setImageResource(Server.typeIcon[Server.typeName.indexOf(s.type)])

            convertView.findViewById<View>(R.id.delete_button).setOnClickListener { adapter?.let { it1 -> DialogDeleteServer().setAdapter(it1).setServerPos(position).show(fragmentManager!!, "DeleteServerDialog") } }

            return convertView
        }
    }
}
